src/SocketManager.cpp

```diff
@@ -39,11 +64,24 @@ void SocketManager::run() {
 					acceptNewConnections(this->fds[i].fd);
 				} else {
 					// read data from the client socket, parse into an HTTP request
-					handleClient(this->fds[i].fd);
+					handleClientRequest(i);
```

Comment:
Rather pass down the entire struct like so:
```cpp
handleClient(this->fds[i]);
```
As you are literally doing this inside the function lol:
```cpp
int fd = this->fds[i];
```